### ISSUE TITLE
Infer and associate images with burst/sequence IDs

### ISSUE THREAD

**Issue by nathanielrindlaub** [tnc-ca-geo/animl-api#77]
Related to #75. 

This seems a bit challenging to do via EXIF data, as how the images represent what burst their associated with differs across makes/manufacturers. 

Perhaps a more universal approach would be to just rely on timestamps, i.e., if an image was taken <1 second after another image, consider it part of a burst with that previous image.

Would need to experiment.

**Comment by nathanielrindlaub** [comment]
Update: we found that by grouping images that were only separated by 2 seconds or less was a really effective way at grouping bursts (see notebook [here](https://github.com/tnc-ca-geo/animl-analytics/blob/main/notebooks/evaluate-mdv5-false-negatives.ipynb)).

I think we should implement this, the question is where. Because you need all the images at once to determine what burst they belong to (and thus can't assign burst when an image gets ingested, for example), and because the available image are dynamic, images are getting added and potentially removed, and because the may not be added in chronological order, I think maybe the best place for this to execute would be when users are exporting data to CSV or COCO.

When you export the data, you're basically taking a snapshot of it, so it's as static and complete as you're going to get. Also, perhaps the `max_delta` (time in seconds) between images that should be grouped together could be a user-supplied integer. That way if there was a reason to group all images that were taken within 5 mins of each other could be considered a "sequence", etc.

**Comment by nathanielrindlaub** [comment]
The first challenge is that we need to be able to sort the images as we read them in from MongoDB first by deployment and then within each deployment chronologically. In theory, something like

```javascript
Image.find(this.query).sort({ deploymentId: 1, dateTimeOriginal: 1 })
```

should do it, but I ran into a mongo error `QueryExceededMemoryLimitNoDiskUseAllowed`. In order to fix this, we'd have to set `AllowedDiskUse` to true (see documentation [here](https://www.mongodb.com/docs/manual/reference/parameters/#mongodb-parameter-param.allowDiskUseByDefault), but that is only [available](https://www.mongodb.com/docs/atlas/reference/free-shared-limitations/#operational-limitations) on dedicated M10 and above instances. 

TL;DR, we may have to wait until upgrading our DB to implement this.

**Comment by nathanielrindlaub** [comment]
See implementation of sequence-level grouping here: https://github.com/tnc-ca-geo/animl-api/blob/31f274f4532d564b24e1c7c0a48e80c10bedaded/src/scripts/analyzeMLSequenceLevel.js#L292

**Comment by nathanielrindlaub** [comment]
We heard from users that it would be really useful to be able to see label counts at the sequence-level in the stats modal. This is not something WI provides either (they just show image-level counts), so could be a nice enhancement.

**Comment by nathanielrindlaub** [comment]
Done. See #318
